terraform/fargate-examples/sqs-dynamic-target-tracking/README.md

```diff
@@ -0,0 +1,80 @@
+# ECS Fargate Autoscaling Queue Processing
+
+This solution blueprint deploys an end to end data processing pipeline using ECS Fargate, Lambda, and SQS. The workload has the following requirements,
+
+1. messages comes to SQS Queue on a regular interval.
+2. each message processing duration varies.
+3. ECS Service needs to scale to handle the messages at scale
+
+For latency-sensitive applications, [AWS guidance describes a common pattern](https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-using-sqs-queue.html) that allows an ECS Service to scale in response to the backlog of an Amazon SQS queue while accounting for the average message processing duration (MPD) and the application’s desired latency.
+
+This blueprint expects samples messages to be available in the SQS Queue. The blueprint provisions a lambda which runs every minute via Event bridge rule with a cron expression. This Eventbridge rule is disabled by default. Once the workload is ready to test, you can enable it to start sending `50` messages to SQS Queue every 1 minute. 
+
+Every message in the SQS Queue has a field to denote the message processes time in seconds. The ECS Task does the following Operations.
+
+1. reads a message from SQS Queue
+2. sleeps for the duration of message processing time
+3. delete the message from SQS Queue
+4. Publish a metric to CloudWatch with value of the message processing time
+
+The ECS Servie is scaled using Target Tracking using a custom metric. The custom metric is ecsBPI and is calculated as follows.
+
+ecsBPI =  ApproximateNumberOfMessages / no of ECS tasks
+
+The target used for ECS Service Target tracking is ecsTargetBPI.
+
+ecsTargetBPI = (maximum acceptable delay) / MPD
+
+
+* Deploy the [core-infra](../core-infra/README.md). Note if you have already deployed the infra then you can reuse it as well.
+* **NOTE:** Codestar notification rules require a **one-time** creation of a service-linked role. Please verify one exists or create the codestar-notification service-linked role.
+  * `aws iam get-role --role-name AWSServiceRoleForCodeStarNotifications`
+
+    ```An error occurred (NoSuchEntity) when calling the GetRole operation: The role with name AWSServiceRoleForCodeStarNotifications cannot be found.```
+  *  If you receive the error above, please create the service-linked role with the `aws cli` below.
+  * `aws iam create-service-linked-role --aws-service-name codestar-notifications.amazonaws.com`
+  * Again, once this is created, you will not have to complete these steps for the other examples.
+* Now you can deploy this blueprint
+```shell
+cd <TOP_DIR>/terraform/fargate-examples/sqs-dynamic-target-tracking
+terraform init
+terraform plan
+terraform apply -auto-approve
+```
+* To test, Go to Eventbridge Console and enable the Eventbridge rule `SQSTestMessageProducer`
```

Comment:
It would be great adding test results and what it means in detail

Reply:
done